cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1045,6 +1050,71 @@ public void setServiceInstances(HttpRequest request, HttpResponder responder,
     }
   }
 
+  /*************************** Queues APIs **************************************/
+  @DELETE
+  @Path("/queues")
+  public synchronized void clearQueues(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId) {
+    // synchronized to avoid a potential race condition here:
+    // 1. the check for state returns that all flows are STOPPED
+    // 2. The API deletes queues because
+    // Between 1. and 2., a flow is started using the /namespaces/{namespace-id}/apps/{app-id}/flows/{flow-id}/start API
+    // Averting this race condition by synchronizing this method. The resource that needs to be locked here is
+    // runtimeService. This should work because the method that is used to start a flow - startStopProgram - is also
+    // synchronized on this.
+    // This synchronization works in HA mode because even in HA mode there is only one leader at a time.
+    try {
+      List<ProgramRecord> flows = listPrograms(Id.Namespace.from(namespaceId), ProgramType.FLOW, store);
+      for (ProgramRecord flow : flows) {
+        String appId = flow.getApp();
+        String flowId = flow.getId();
+        Id.Program programId = Id.Program.from(namespaceId, appId, flowId);
+        ProgramStatus status = getProgramStatus(programId, ProgramType.FLOW);
+        if (!"STOPPED".equals(status.getStatus())) {
+          responder.sendString(HttpResponseStatus.FORBIDDEN,
+                               String.format("Flow '%s' from application '%s' in namespace '%s' is running, " +
+                                               "please stop it first.", flowId, appId, namespaceId));
+          return;
+        }
+      }
+    } catch (Exception e) {
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+      return;
+    }
+    // delete process metrics that are used to calculate the queue size (process.events.pending metric name)
+    // TODO: CDAP-1184 Implement metrics deletion once metrics APIs accept namespace
+    clear(request, responder, namespaceId, ToClear.QUEUES);
+  }
+
+  /**
+   * Protected until /v3 APIs for streams are added
+   */
+  protected static enum ToClear {
+    QUEUES, STREAMS
```

Comment:
Removed the ``clear`` method and the ``ToClear`` enum altogether. Just used ``streamAdmin`` and ``queueAdmin`` methods directly in the respective APIs.